aries_vcx/src/messages/a2a/message_type.rs

```diff
@@ -52,6 +52,73 @@ impl Serialize for MessageType {
 
 impl std::string::ToString for MessageType {
     fn to_string(&self) -> String {
-        format!("{};spec/{}/{}/{}", self.did, self.family.to_string(), self.version, self.type_)
+        format!("{}/{}/{}/{}", self.prefix, self.family.to_string(), self.version, self.msg_type)
+    }
+}
+
+
+pub fn parse_message_type_legacy(message_type: &str) -> Option<MessageType> {
```

Comment:
FYI for review:

The code in `parse_message_type_legacy` and `parse_message_type` was copied here from `use agency_client::message_type::parse_message_type` to make it possible to have different behaviour for aries messages and agency messages (`@type` encoding changes in client2agency protoco are not desirable at the moment).

Reply:
Also it feels cleaner that this module does not depend on `agency_client`

Reply:
When in future we rewrite client2agency protocol to be aries comptabile, we can extract `message_type`, `message_family` etc.  and reuse it from both `aries-vcx` and `agency-client`

Reply:
Why are they not desirable? I think it is desirable that forward messages are treated the same as client-to-client messages, but the problem is that agency does not support both formats ATM, correct? If that's so, we should implement the new format in the agency ASAP and get rid of this duplication by extracting the types into a separate module used both by `agency-client` and by `aries-vcx` - do you agree?

Reply:
I agree, it should be done. To clarify, I didn't want to make the scope of the PR too broad, aiming primarily fixing the issue at hand (e2e aries interoperability issue).